client/signup/steps/site-info-collection/index.tsx

```diff
@@ -0,0 +1,277 @@
+					value={ formValues.siteDescription }
+					name="siteDescription"
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: formValues.siteDescription,
+		},
+		{
+			title: translate( '%d. Social Media Profiles', {
+				args: [ 3 ],
+				comment: 'This is the serial number: 3',
+			} ),
+			component: (
+				<SocialMediaProfiles
+					facebookProps={ {
+						value: formValues.facebookUrl,
+						name: 'facebookUrl',
+					} }
+					twitterProps={ {
+						value: formValues.twitterUrl,
+						name: 'twitterUrl',
+					} }
+					linkedinProps={ {
+						value: formValues.linkedinUrl,
+						name: 'linkedinUrl',
+					} }
+					instagramProps={ {
+						value: formValues.instagramUrl,
+						name: 'instagramUrl',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+		{
+			title: translate( '%d. Contact Information', {
+				args: [ 4 ],
+				comment: 'This is the serial number: 4',
+			} ),
+			component: (
+				<ContactInformation
+					displayEmailProps={ {
+						value: formValues.displayEmail,
+						name: 'displayEmail',
+					} }
+					displayPhoneProps={ {
+						value: formValues.displayPhone,
+						name: 'displayPhone',
+					} }
+					displayAddressProps={ {
+						value: formValues.displayAddress,
+						name: 'displayAddress',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+	];
+
+	const isSectionAtIndexTouchedInitialState: Record< string, boolean > = {};
+	for ( let i = 0; i <= currentIndex; i++ ) {
+		isSectionAtIndexTouchedInitialState[ `${ i }` ] = true;
+	}
+
+	const [ isSectionAtIndexATouched, setIsSectionAtIndexATouched ] = useState<
+		Record< string, boolean >
+	>( isSectionAtIndexTouchedInitialState );
+
+	const runValidation = ( validator: ValidatorFunction ) => {
+		const validationResult = validator( formValues );
+		setFormErrors( {
+			...formErrors,
+			...validationResult.errors,
+		} );
+		return validationResult;
+	};
+
+	const submitStep = () => {
+		for ( let index = 0; index < sections.length; index++ ) {
+			const section = sections[ index ];
+			if ( section.validate ) {
+				const validationResult = runValidation( section.validate );
+				if ( ! validationResult.result ) {
+					dispatch( updateSiteInfoCurrentIndex( index ) );
+					return;
+				}
+			}
+		}
+
+		const step = {
+			stepName,
+			...additionalStepData,
+		};
+		submitSignupStep( step, {
+			...formValues,
+		} );
+		goToNextStep();
+	};
+
+	const onOpen = ( currentIndex: number ) => {
+		dispatch( updateSiteInfoCurrentIndex( currentIndex ) );
+		dispatch( updateSiteInfoValues( formValues ) );
+		setIsSectionAtIndexATouched( { ...isSectionAtIndexATouched, [ `${ currentIndex }` ]: true } );
+	};
+
+	const onNext = ( validator?: ValidatorFunction ) => {
+		dispatch( updateSiteInfoValues( formValues ) );
+
+		if ( validator ) {
+			const validationResult = runValidation( validator );
```

Comment:
nitpick!
This feels like an unnecessary indirection, why not call the function and set the error result right here?
Also I felt confused when runValidation also populated an errors.

Reply:
Can you help me understand your question?

> why not call the function and set the error result right here?

Which function do you mean?

Reply:
Ah sorry I mean `runValidation`
```suggestion
			const validationResult = validator( formValues );
		        setFormErrors( {
			        ...formErrors,
			        ...validationResult.errors,
		        } );
```

Reply:
The `runValidation` function does two things:
1. Execute the validator function.
2. Set error states in redux.

Since the same code was repeated in two places ( `submitStep` and `onNext`), I abstracted this into a separate function.

Reply:
I see!
DRY can be nice win for removing duplication, most of the time, but I feel that this might be a bit overkill. And like anything DRY is a tradeoff which must be carefully evaluated as with all things. Some further reading!
https://www.deconstructconf.com/2019/dan-abramov-the-wet-codebase

Regarding the two things, 
1 - Yes, executing validator function is calling`validator` function with a parameter.
2 - But Naming a function `runValidator` and then doing other things inside that function can lead to confusion. It feels like the name of the function is lying to me when it says it's just going to `run the validator` when in fact it's also going to set some state. 😅 

The obvious fix I see of course is to just remove the DRY abstractions and live with the duplication which I think can be a win for readability. This is  just what I felt when reading the code and trying understand what's happening. But, another possible option is to live with this DRY abstraction by coming up with a better name for that function, something like `validateFieldsAndInitializeFieldErrors` 🤔 
or something better if you can come up with something 💡

Reply:
> https://www.deconstructconf.com/2019/dan-abramov-the-wet-codebase

Thanks for sharing the article. It's a great read!

In this case, the `validator` function is pure and does not cause side effects. So, running just this function without the subsequent call to `setFormErrors` does not make sense. A call to the `validator` will almost always be followed by a call to `setFormErrors`. For this reason, I have added these two things to the same function. 

>  But Naming a function runValidator and then doing other things inside that function can lead to confusion. 

Agreed, the naming here could have been better. I have renamed this function to `runValidatorAndSetFormErrors` in 0f5bcd2fdfe26505aed59340a300930ed3242051.